src/Behat/Behat/Tester/Cli/RerunController.php

```diff
@@ -64,6 +67,11 @@ public function __construct(EventDispatcherInterface $eventDispatcher, $cachePat
         $this->basepath = $basepath;
     }
 
+    public function setResultInterpreter(ResultInterpreter $resultInterpreter)
```

Comment:
As this class is final, why not changing the `__construct()`?

Reply:
Even though it is final, there is the possibility that some extension is using it and creating an instance. We prefer to be careful and not introduce a BC break if possible. In a future major version we can probably move this to the constructor

Reply:
@carlos-granados I've been thinking about this pattern some more and actually using the setter injection we still have a technical BC break if people are using the class without providing the interpreter. It will just come at runtime when `resultInterpreter` is undefined, instead of when they create the class.

There is a way to make this one fully BC by doing `if (isset($this->result interpreter))` and falling back to the old logic if not.

But I wonder if it's actually worth the indirection, or if we should just work on the basis that these classes are internal to Behat and not part of the public API (even though that has never been explicitly said), and go for the neater/more visible solution of passing this in the constructor and highlighting it in the changelog...

What do you think?

Reply:
@acoulton mmm, yeah you may be right. I did a search on Github for RerunController and Behat and the only extension that I could find that did something with it was the https://github.com/Daniel-Marynicz/BehatParallelExtension extension. But this extension just decorates the controller, so should continue working fine if we modify the controller constructor.
Let me update the PR with this change

Reply:
@acoulton updated

Reply:
@carlos-granados thanks - I've had a look at that extension and agree this shouldn't affect it. And that also proves it's possible to decorate our controllers with the DI container config, rather than creating them manually, so hopefully any other extensions will be following that pattern.